### ISSUE TITLE
关于前背景权重的计算

### ISSUE THREAD

**Issue by lsh-med** [Henryjiepanli/DAD#3]
def calculate_dynamic_weights(self, features, mask):
        mask = F.interpolate(mask, size=features.size()[2:], mode='bilinear', align_corners=True)
        mask = mask > 0.5
        intensity = torch.sum(features, dim=1, keepdim=True)
        intensity = intensity / torch.max(intensity)
        weights = intensity * mask.float()
        return weights

    def calculate_prototype_with_weights(self, features, mask):
        weights = self.calculate_dynamic_weights(features, mask)
        masked_features = features * weights
        prototype = masked_features.sum(dim=(2, 3)) / (weights.sum(dim=(2, 3)) + 1e-8)
        return prototype

作者您好，我想问下对于权重得计算过程中：intensity = torch.sum(features, dim=1, keepdim=True)
                                                                          intensity = intensity / torch.max(intensity)  该操作的目的是什么，通过对不同通道的特征进行求和和归一化处理，再将其与mask的硬注意力特征图像进行相乘，这一系列操作应该怎么理解。
并且 prototype = masked_features.sum(dim=(2, 3)) / (weights.sum(dim=(2, 3)) + 1e-8) 这一步的操作也不是很懂，希望作者可以解惑一下

**Comment by Henryjiepanli** [comment]
你好，感谢对我们方法的关注。根据审稿人意见，我们方法进行了较大的调整，所以现在模块的方法和axirv论文已经发生了很大的改变。关于这个模块，本质上就是动态的计算前景与背景的原型。如果有不理解的地方可以参考一下few-shot中常见的原型策略。

**Comment by lsh-med** [comment]
> 你好，感谢对我们方法的关注。根据审稿人意见，我们方法进行了较大的调整，所以现在模块的方法和axirv论文已经发生了很大的改变。关于这个模块，本质上就是动态的计算前景与背景的原型。如果有不理解的地方可以参考一下few-shot中常见的原型策略。

那关于论文中的具体操作，可以分享一下吗，对您的方法很感兴趣

**Comment by Henryjiepanli** [comment]
你好，论文暂时还没有在arXiv更新，等后续更新，我会第一时间告知，谢谢关注

2024-12-12 23:18:59 "lsh-med" ***@***.***> 写道：

你好，感谢对我们方法的关注。根据审稿人意见，我们方法进行了较大的调整，所以现在模块的方法和axirv论文已经发生了很大的改变。关于这个模块，本质上就是动态的计算前景与背景的原型。如果有不理解的地方可以参考一下few-shot中常见的原型策略。

那关于论文中的具体操作，可以分享一下吗，对您的方法很感兴趣
